pkg/execution/svccatadmission/internaldns.go

```diff
@@ -78,6 +78,20 @@ func InternalDNSAdmitFunc(ctx context.Context, microsServerClient microsServerCl
 		return nil, errors.Errorf("error fetching service central data for serviceName %q", serviceName)
 	}
 
+	if serviceCentralData == nil {
```

Comment:
@marciogh please focus on this change

Reply:
Did we run into this issue recently? This could only happen when someone delete a service from Service Central without deleting SDs from clusters first.

Reply:
if `getServiceData` uses our ServiceCentral client, it returns an error for 404, not nil. So you could check for `if servicecentral.IsNotFound(err)` or similar.

Reply:
The code doesn't actually handle this case correctly. If we do hit this we npe and block all changes to ingress for everyone.

Reply:
The `getServiceData` function use list with a query. That would can return 0 results.

Reply:
Starting to look at this now

Reply:
https://sdog.jira-dev.com/browse/MICROS-6941

Reply:
@amckague @nilebox sorry, I don't see what's wrong here, need advice:

> if `getServiceData` uses our ServiceCentral client, it returns an error for 404, not nil. So you could check for `if servicecentral.IsNotFound(err)` or similar.

If I can't find the service on SC, I can't validate the domain, therefore, can't allow it to be created. So there's no real difference if I check for error or if serviceData is nil.

> The `getServiceData` function use list with a query. That would can return 0 results. (and a return result of nil, nil)

If I get nil, nil means the service couldn't be found. Again, I have to fail because don't have required information to allow the ingress.

>The code doesn't actually handle this case correctly. If we do hit this we npe and block all changes to ingress for everyone.

Where exactly the npe would happen?